### ISSUE TITLE
Support pipelines without deploy step

### ISSUE THREAD

**Issue by mxey** [GoogleContainerTools/skaffold#1276]
I'd like to standardize my CI pipelines on Skaffold (for consistency, and because `skaffold dev` that automatically runs tests is useful). I can already use Skaffold without building anything, if I just want to deploy some resources.  However, it's not possible to leave out a deployer, because there's always the default kubectl deployer.

The workaround I came up with is using an empty manifest file with the kubectl deployer:

```yaml
deploy:
  kubectl:
    manifests:
    - manifest.yaml
```

This works, but also logs a warning `WARN[0004] image […] is not used by the deployment `

**Comment by mxey** [comment]
@tejal29 I can use `skaffold build` in a pipeline, but I think `skaffold dev` always wants to deploy?

**Comment by tstromberg** [comment]
I'm closing this issue as it hasn't seen activity in awhile, and it's unclear if this issue still exists. If this issue does continue to exist in the most recent release of Skaffold, please feel free to add a comment and we'll re-open it.

If someone sees a similar issue, please create a new issue, but do include a link to this issue if possible.

Thank you for opening the issue!

**Comment by dgageot** [comment]
@mxey I'm not sure I've got a good understanding of your use case. Could you give use more details?

What I think I understand is that you'd like to use a flavour of `skaffold dev` that wouldn't deploy. Could you tell us what it would be used for?

**Comment by mxey** [comment]
@dgageot triggering rebuild and test of a Docker image when source files change

**Comment by tejal29** [comment]
@mxey, so yiu are using `skaffold dev` to just build your images. 
As part of your development workflow, you dont want to deploy them to a cluster?

Just curious, 
How do developers test (other than unit test and container structure tests) the built container application? do you use any other tools to deploy?

**Comment by dgageot** [comment]
What about a `skaffold build --watch` command?

**Comment by felixtran39** [comment]
Since we still want to support users who are using the default deployer, it would probably be best to support a command like `skaffold dev --no-deploy`. I'm curious about how drastic the concerns are regarding the `if-else` checks that need to be added. @tejal29 WDYT?

**Comment by nkubala** [comment]
@felixtran39 yeah i think if we're going to support this we should do it through a flag, since we don't want to mess with the default behavior we have right now
